jets.hxx

```diff
@@ -0,0 +1,105 @@
+                         const std::string &jet_eta, const std::string &jet_phi,
+                         const std::string &p4_1, const std::string &p4_2,
+                         const float deltaRmin) {
+    auto df1 = df.Define(
+        output_col,
+        [deltaRmin](const ROOT::RVec<float> &jet_eta,
+                    const ROOT::RVec<float> &jet_phi,
+                    const ROOT::Math::PtEtaPhiMVector &p4_1,
+                    ROOT::Math::PtEtaPhiMVector &p4_2) {
+            Logger::get("VetoOverlappingJets")->debug("Checking jets");
+            ROOT::RVec<int> mask(jet_eta.size(), 1);
+            for (std::size_t idx = 0; idx < mask.size(); ++idx) {
+                ROOT::Math::RhoEtaPhiVectorF jet(0, jet_eta[idx], jet_phi[idx]);
+                Logger::get("VetoOverlappingJets")
+                    ->debug("Jet:  Eta: {} Phi: {} ", jet.Eta(), jet.Phi());
+                Logger::get("VetoOverlappingJets")
+                    ->debug("Letpon 1 {}:  Eta: {} Phi: {}, Pt{}", p4_1,
+                            p4_1.Eta(), p4_1.Phi(), p4_1.Pt());
+                Logger::get("VetoOverlappingJets")
+                    ->debug("Lepton 2 {}:  Eta: {} Phi: {}, Pt{}", p4_2,
+                            p4_2.Eta(), p4_2.Phi(), p4_2.Pt());
+                auto deltaR_1 = ROOT::Math::VectorUtil::DeltaR(jet, p4_1);
+                auto deltaR_2 = ROOT::Math::VectorUtil::DeltaR(jet, p4_2);
+                Logger::get("VetoOverlappingJets")
+                    ->debug("DeltaR 1 {}", deltaR_1);
+                Logger::get("VetoOverlappingJets")
+                    ->debug("DeltaR 2 {}", deltaR_2);
+                mask[idx] = (deltaR_1 > deltaRmin && deltaR_2 > deltaRmin);
+            }
+            return mask;
+        },
+        {jet_eta, jet_phi, p4_1, p4_2});
+    return df1;
+}
+
+auto OrderJetsByPt(auto df, const std::string &output_col,
+                   const std::string &jet_pt, const std::string &jetmask) {
+    auto df1 = df.Define(
+        output_col,
+        [](const ROOT::RVec<int> &jetmask, const ROOT::RVec<float> &jet_pt) {
+            Logger::get("OrderJetsByPt")->debug("Ordering good jets by pt");
+            Logger::get("OrderJetsByPt")->debug("Jetpt before {}", jet_pt);
+            Logger::get("OrderJetsByPt")->debug("Mask {}", jetmask);
+            auto good_jets_pt =
+                ROOT::VecOps::Where(jetmask > 0, jet_pt, (float)0.);
+            Logger::get("OrderJetsByPt")->debug("Jetpt after {}", good_jets_pt);
+            // we have to convert the result into an RVec of ints since argsort
+            // gives back an unsigned long vector
+            auto temp =
+                ROOT::VecOps::Argsort(ROOT::VecOps::Nonzero(good_jets_pt));
+            Logger::get("OrderJetsByPt")->debug("jet Indices {}", temp);
+            ROOT::RVec<int> result(temp.size());
+            std::transform(temp.begin(), temp.end(), result.begin(),
```

Comment:
This part did cause some crashes, when i first did this, but i guess this is fixed now ?

Reply:
Right, this is what I mentioned in my private message already. The transform function does not check the size of the target vector (it was initialized empty before). Now it is initialized with correct size. There is also a method to increase size within transform (inserting some function similar to push_back instead of the plain iterator), but I think it's best to reserve the correct amount of memory right away.